src/tunnel/server.rs

```diff
@@ -100,6 +104,10 @@ async fn run_tunnel(
             static SERVERS: Lazy<Mutex<HashMap<(Host<String>, u16), mpsc::Receiver<Item>>>> =
                 Lazy::new(|| Mutex::new(HashMap::with_capacity(0)));
 
+            if server_config.http_proxy.is_some() {
```

Comment:
You can remove the error check. Reverse tunneling is not going to use the HTTP proxy as, it is incoming traffic to the server and after is going throught the tunnel